src/libsyncengine/update_detection/file_system_observer/remotefilesystemobserverworker.cpp

```diff
@@ -688,49 +688,49 @@ ExitCode RemoteFileSystemObserverWorker::processAction(const SyncName &usedName,
     }
 
 
-    return ExitCodeOk;
+    return ExitCode::Ok;
 }
 
 ExitCode RemoteFileSystemObserverWorker::checkRightsAndUpdateItem(const NodeId &nodeId, bool &hasRights,
                                                                   SnapshotItem &snapshotItem) {
     GetFileInfoJob job(_syncPal->driveDbId(), nodeId);
     job.runSynchronously();
-    if (job.hasHttpError() || job.exitCode() != ExitCodeOk) {
+    if (job.hasHttpError() || job.exitCode() != ExitCode::Ok) {
         if (job.getStatusCode() == Poco::Net::HTTPResponse::HTTP_FORBIDDEN ||
             job.getStatusCode() == Poco::Net::HTTPResponse::HTTP_NOT_FOUND) {
             hasRights = false;
-            return ExitCodeOk;
+            return ExitCode::Ok;
         }
 
         LOGW_SYNCPAL_WARN(_logger, L"Error while determining access rights on item: "
                                        << SyncName2WStr(snapshotItem.name()).c_str() << L" ("
                                        << Utility::s2ws(snapshotItem.id()).c_str() << L")");
         invalidateSnapshot();
-        return ExitCodeBackError;
+        return ExitCode::BackError;
     }
 
     snapshotItem.setCreatedAt(job.creationTime());
     snapshotItem.setLastModified(job.modtime());
     snapshotItem.setSize(job.size());
     hasRights = true;
-    return ExitCodeOk;
+    return ExitCode::Ok;
 }
 
 bool RemoteFileSystemObserverWorker::hasUnsupportedCharacters(const SyncName &name, const NodeId &nodeId, NodeType type) {
 #ifdef __APPLE__
     // Check that the name doesn't contain a character not yet supported by the filesystem (ex: U+1FA77 on pre macOS 13.4)
     bool valid = false;
-    if (type == NodeTypeFile) {
+    if (type == NodeType::File) {
         valid = CommonUtility::fileNameIsValid(name);
-    } else if (type == NodeTypeDirectory) {
+    } else if (type == NodeType::Directory) {
         valid = CommonUtility::dirNameIsValid(name);
     }
 
     if (!valid) {
         LOGW_SYNCPAL_DEBUG(_logger, L"The file/directory name contains a character not yet supported by the filesystem "
                                         << SyncName2WStr(name).c_str() << L". Item is ignored.");
 
-        Error err(_syncPal->syncDbId(), "", nodeId, type, name, ConflictTypeNone, InconsistencyTypeNotYetSupportedChar);
+        Error err(_syncPal->syncDbId(), "", nodeId, type, name, ConflictType::None, InconsistencyTypeNotYetSupportedChar);
```

Comment:
What about `InconsistencyType::NotYetSupportedChar`?

Reply:
These classes can't be easily converted to enum classes due to the numerous bitwise operations involved. I will take a deeper look to see if I can find an efficient way to convert them to enum classes without making the code unreadable.

Reply:
You can check these 3 commits [5470bb5](https://github.com/Infomaniak/desktop-kDrive/pull/174/commits/5470bb59611406b776340857d97b1652f36be8e9), [c6f3ed3](https://github.com/Infomaniak/desktop-kDrive/pull/174/commits/c6f3ed3cf653294b3238998cccb21314205a88c2), [78afc22](https://github.com/Infomaniak/desktop-kDrive/pull/174/commits/78afc22338b62d49b85ebc838283fcf3202fea63).